src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -7,297 +7,405 @@
-                } else if (factor instanceof BranchFlowPerPSTAngle) {
-                    BranchFlowPerPSTAngle pstAngleFactor = (BranchFlowPerPSTAngle) factor;
-                    sensitivityValues.add(createBranchSensitivityValue(lfNetwork, equationSystem, pstAngleFactor.getFunction().getBranchId(),
-                            factor, factorGroup, states, functionReferenceByBranch));
-                } else {
-                    throw new UnsupportedOperationException("Factor type '" + factor.getClass().getSimpleName() + "' not yet supported");
-                }
+    protected List<SensitivityValue> calculateSensitivityValues(List<SensitivityFactorGroup> factorGroups,
+                                                                DenseMatrix states, List<ComputedContingencyElement> contingencyElements) {
+        List<SensitivityValue> sensitivityValuesContingencies = new ArrayList<>(factorGroups.stream().mapToInt(group -> group.getFactors().size()).sum());
+        for (SensitivityFactorGroup factorGroup : factorGroups) {
+            setAlphas(contingencyElements, factorGroup, states);
+            for (SensitivityFactorWrapped factor : factorGroup.getFactors()) {
+                sensitivityValuesContingencies.add(createBranchSensitivityValue(factor, factorGroup, states, contingencyElements));
             }
         }
+        return sensitivityValuesContingencies;
+    }
 
-        return sensitivityValues;
+    private void setAlphas(List<ComputedContingencyElement> contingencyElements, SensitivityFactorGroup sensitivityFactorGroup, DenseMatrix states) {
+        ComputedContingencyElement.setContingencyIndexes(contingencyElements);
+        if (contingencyElements.size() == 1) {
+            ComputedContingencyElement element = contingencyElements.get(0);
+            LfBranch lfBranch = element.getLfBranch();
+            ClosedBranchSide1DcFlowEquationTerm p1 = element.getBranchEquation();
+            // we solve a*alpha = b
+            double a = lfBranch.getPiModel().getX() / PerUnit.SB - (states.get(p1.getVariables().get(0).getRow(), element.getGlobalIndex())
+                             - states.get(p1.getVariables().get(1).getRow(), element.getGlobalIndex()));
+            double b = states.get(p1.getVariables().get(0).getRow(), sensitivityFactorGroup.getIndex())
+                       - states.get(p1.getVariables().get(1).getRow(), sensitivityFactorGroup.getIndex());
+            element.setAlpha(b / a);
+        } else {
+            DenseMatrix rhs = new DenseMatrix(contingencyElements.size(), 1);
+            DenseMatrix matrix = new DenseMatrix(contingencyElements.size(), contingencyElements.size());
+            for (ComputedContingencyElement element : contingencyElements) {
+                if (!element.getElement().getType().equals(ContingencyElementType.BRANCH)) {
+                    throw new UnsupportedOperationException("Only contingency on a branch is yet supported");
```

Comment:
You check contingencies in the very beginning of your analysis. Do we really need this exception?